src/workspaces/features/DeleteWorkspaceDialog/DeleteWorkspaceDialog.tsx

```diff
@@ -0,0 +1,63 @@
+import Button from "core/components/Button";
+import Dialog from "core/components/Dialog";
+import Spinner from "core/components/Spinner";
+import { useState } from "react";
+import { useTranslation } from "react-i18next";
+import { useRouter } from "next/router";
+import { useDeleteWorkspaceMutation } from "workspaces/graphql/mutations.generated";
+import { Workspace } from "graphql-types";
+import useCacheKey from "core/hooks/useCacheKey";
+
+type DeleteWorkspaceDialogProps = {
+  onClose(): void;
+  open: boolean;
+  workspace: Pick<Workspace, "id" | "name">;
```

Comment:
It's better to create a fragment for the DeleteWorkspaceDialog and include this fragment in the calling parent.

Reply:
Okay